internal/planner/scan.go

```diff
@@ -152,46 +153,13 @@ func (n *scanNode) addField(field client.FieldDefinition) {
 	}
 }
 
-func (scan *scanNode) initFetcher(
-	cid immutable.Option[string],
-	index immutable.Option[client.IndexDescription],
-) {
+func (scan *scanNode) initFetcher(cid immutable.Option[string]) {
 	var f fetcher.Fetcher
 	if cid.HasValue() {
 		f = new(fetcher.VersionedFetcher)
 	} else {
 		f = fetcher.NewDocumentFetcher()
 
-		if index.HasValue() {
```

Comment:
praise: I was worried you'd have to put this back when you moved the index prop to init in the fetcher.  Thank you very much for cleaning this up.